DDS installation fails with missing boost symbol

Dear developers,
I got the following error when I try to install FairSoft via spack. The problem is connected to the DDS installation with missing symbols from boost: "boost::systemn::detail::generic_category_ncx()"

Here is the complete output:

```
Installing dds
==> Searching for binary cache of dds
==> Warning: No Spack mirrors are currently configured
==> No binary for dds found: installing from source
==> Warning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check clang@10.0.1-apple for further details]
==> Cloning git repository: https://github.com/FairRootGroup/DDS at commit 7499753bdec9b5ed2468a712e57c5578ca25e7a6
==> No checksum needed when fetching with git
==> Already staged spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson in /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson
==> Applied patch /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/packages/dds/fix_wn_bin_2.4.patch
==> Building dds [CMakePackage]
==> Executing phase: 'cmake'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4' 'all' 'wn_bin'

17 errors found in build log:
     89     cd /private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-build/MiscCommon/dds_n
            cf && /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/spack/lib/spack/env/clang/clang++  -DBOOST_ALL_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_PROGRAM_OPTIONS_
            DYN_LINK -Ddds_ncf_EXPORTS -I/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-src/
            MiscCommon/dds_ncf/src -I/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-src/Misc
            Common -isystem /usr/local/include  -std=c++11 -stdlib=libc++ -Wthread-safety -O2 -g -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.p
            latform/Developer/SDKs/MacOSX10.14.sdk -fPIC   -o CMakeFiles/dds_ncf.dir/src/ncf.cpp.o -c /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/sp
            ack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-src/MiscCommon/dds_ncf/src/ncf.cpp
     90     [  3%] Building CXX object dds-user-defaults/CMakeFiles/dds-user-defaults.dir/src/main.cpp.o
     91     cd /private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-build/dds-user-default
            s && /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/spack/lib/spack/env/clang/clang++  -DBOOST_ALL_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_PROGRAM_OPTIONS_D
            YN_LINK -DBOOST_SYSTEM_DYN_LINK -I/private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefs
            on/spack-build/dds-user-defaults -I/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spac
            k-src/MiscCommon -isystem /usr/local/include  -std=c++11 -stdlib=libc++ -Wthread-safety -O2 -g -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platform
            s/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk   -o CMakeFiles/dds-user-defaults.dir/src/main.cpp.o -c /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns
            /spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-src/dds-user-defaults/src/main.cpp
     92     [  3%] Linking CXX executable dds-daemonize
     93     cd /private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-build/dds-daemonize &&
             /Users/tstockmanns/.spack/install_tree/darwin-mojave-x86_64/clang-10.0.1-apple/cmake-3.16.2-4vwklyt5nxixy7z2xyuipcnvmqn57pfj/bin/cmake -E cmake_link_script CMakeFil
            es/dds-daemonize.dir/link.txt --verbose=1
     94     /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/spack/lib/spack/env/clang/clang++   -std=c++11 -stdlib=libc++ -Wthread-safety -O2 -g -DNDEBUG -isysroot /Applica
            tions/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  -lc++ CMakeFile
            s/dds-daemonize.dir/src/main.cpp.o  -o dds-daemonize  /usr/local/lib/libboost_program_options-mt.dylib /usr/local/lib/libboost_system-mt.dylib /usr/local/lib/libboos
            t_filesystem-mt.dylib
  \>> 95     Undefined symbols for architecture x86_64:
     96       "boost::system::detail::system_category_ncx()", referenced from:
     97           ___cxx_global_var_init.28 in main.cpp.o
     98           ___cxx_global_var_init.32 in main.cpp.o
     99           ___cxx_global_var_init.37 in main.cpp.o
     100          __GLOBAL__sub_I_main.cpp in main.cpp.o
     101      "boost::system::detail::generic_category_ncx()", referenced from:
     102          boost::system::error_category::std_category::equivalent(int, std::__1::error_condition const&) const in main.cpp.o
     103          boost::system::error_category::std_category::equivalent(std::__1::error_code const&, int) const in main.cpp.o
     104    ld: symbol(s) not found for architecture x86_64
  \>> 105    clang: error: linker command failed with exit code 1 (use -v to see invocation)
  \>> 106    make[2]: *** [dds-daemonize/dds-daemonize] Error 1
  \>> 107    make[1]: *** [dds-daemonize/CMakeFiles/dds-daemonize.dir/all] Error 2
     108    make[1]: *** Waiting for unfinished jobs....
     109    [  4%] Building CXX object dds-user-defaults/CMakeFiles/dds-user-defaults.dir/src/UserDefaults.cpp.o
     110    cd /private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-build/dds-user-default
            s && /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/spack/lib/spack/env/clang/clang++  -DBOOST_ALL_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_PROGRAM_OPTIONS_D
            YN_LINK -DBOOST_SYSTEM_DYN_LINK -I/private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefs
            on/spack-build/dds-user-defaults -I/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spac
            k-src/MiscCommon -isystem /usr/local/include  -std=c++11 -stdlib=libc++ -Wthread-safety -O2 -g -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platform
            s/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk   -o CMakeFiles/dds-user-defaults.dir/src/UserDefaults.cpp.o -c /var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tst
            ockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-src/dds-user-defaults/src/UserDefaults.cpp
     111    [  5%] Linking CXX shared library libdds_ncf.dylib
     112    cd /private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-build/MiscCommon/dds_n
            cf && /Users/tstockmanns/.spack/install_tree/darwin-mojave-x86_64/clang-10.0.1-apple/cmake-3.16.2-4vwklyt5nxixy7z2xyuipcnvmqn57pfj/bin/cmake -E cmake_link_script CMa
            keFiles/dds_ncf.dir/link.txt --verbose=1
     113    /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/spack/lib/spack/env/clang/clang++  -std=c++11 -stdlib=libc++ -Wthread-safety -O2 -g -DNDEBUG -isysroot /Applicat
            ions/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -dynamiclib -Wl,-headerpad_max_install_names  -o libdds_ncf.dylib -install
            _name @rpath/libdds_ncf.dylib CMakeFiles/dds_ncf.dir/src/ncf.cpp.o  /usr/local/lib/libboost_program_options-mt.dylib /usr/local/lib/libboost_filesystem-mt.dylib
  \>> 114    Undefined symbols for architecture x86_64:
     115      "boost::system::detail::system_category_ncx()", referenced from:
     116          ___cxx_global_var_init.29 in ncf.cpp.o
     117          ___cxx_global_var_init.34 in ncf.cpp.o
     118          ___cxx_global_var_init.35 in ncf.cpp.o
     119          __GLOBAL__sub_I_ncf.cpp in ncf.cpp.o
     120      "boost::system::detail::generic_category_ncx()", referenced from:
     121          boost::system::error_category::std_category::equivalent(int, std::__1::error_condition const&) const in ncf.cpp.o
     122          boost::system::error_category::std_category::equivalent(std::__1::error_code const&, int) const in ncf.cpp.o
     123    ld: symbol(s) not found for architecture x86_64
  \>> 124    clang: error: linker command failed with exit code 1 (use -v to see invocation)
  \>> 125    make[2]: *** [MiscCommon/dds_ncf/libdds_ncf.dylib] Error 1
  \>> 126    make[1]: *** [MiscCommon/dds_ncf/CMakeFiles/dds_ncf.dir/all] Error 2
     127    [  6%] Linking CXX shared library libdds-user-defaults.dylib
     128    cd /private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-build/dds-user-default
            s && /Users/tstockmanns/.spack/install_tree/darwin-mojave-x86_64/clang-10.0.1-apple/cmake-3.16.2-4vwklyt5nxixy7z2xyuipcnvmqn57pfj/bin/cmake -E cmake_link_script CMak
            eFiles/dds_user_defaults_lib.dir/link.txt --verbose=1
     129    /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/spack/lib/spack/env/clang/clang++  -std=c++11 -stdlib=libc++ -Wthread-safety -O2 -g -DNDEBUG -isysroot /Applicat
            ions/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -dynamiclib -Wl,-headerpad_max_install_names  -o libdds-user-defaults.dyli
            b -install_name @rpath/libdds-user-defaults.dylib CMakeFiles/dds_user_defaults_lib.dir/src/UserDefaults.cpp.o  /usr/local/lib/libboost_program_options-mt.dylib /usr/
            local/lib/libboost_system-mt.dylib /usr/local/lib/libboost_filesystem-mt.dylib
  \>> 130    Undefined symbols for architecture x86_64:
     131      "boost::system::detail::system_category_ncx()", referenced from:
     132          ___cxx_global_var_init.92 in UserDefaults.cpp.o
     133          ___cxx_global_var_init.96 in UserDefaults.cpp.o
     134          ___cxx_global_var_init.101 in UserDefaults.cpp.o
     135          __GLOBAL__sub_I_UserDefaults.cpp in UserDefaults.cpp.o
     136      "boost::system::detail::generic_category_ncx()", referenced from:
     137          boost::system::error_category::std_category::equivalent(int, std::__1::error_condition const&) const in UserDefaults.cpp.o
     138          boost::system::error_category::std_category::equivalent(std::__1::error_code const&, int) const in UserDefaults.cpp.o
     139    ld: symbol(s) not found for architecture x86_64
  \>> 140    clang: error: linker command failed with exit code 1 (use -v to see invocation)
  \>> 141    make[2]: *** [dds-user-defaults/libdds-user-defaults.dylib] Error 1
  \>> 142    make[1]: *** [dds-user-defaults/CMakeFiles/dds_user_defaults_lib.dir/all] Error 2
     143    [  7%] Linking CXX executable dds-user-defaults
     144    cd /private/var/folders/_1/1x665w457cgdjljzc_t7_l5m0000gn/T/tstockmanns/spack-stage/spack-stage-dds-2.4-ex2o5wuprpvg6h2qkjpfjqgpvtwefson/spack-build/dds-user-default
            s && /Users/tstockmanns/.spack/install_tree/darwin-mojave-x86_64/clang-10.0.1-apple/cmake-3.16.2-4vwklyt5nxixy7z2xyuipcnvmqn57pfj/bin/cmake -E cmake_link_script CMak
            eFiles/dds-user-defaults.dir/link.txt --verbose=1
     145    /Users/tstockmanns/PandaSoftware/FairSoft/dev/source/spack/lib/spack/env/clang/clang++   -std=c++11 -stdlib=libc++ -Wthread-safety -O2 -g -DNDEBUG -isysroot /Applica
            tions/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  -lc++ CMakeFile
            s/dds-user-defaults.dir/src/main.cpp.o CMakeFiles/dds-user-defaults.dir/src/UserDefaults.cpp.o  -o dds-user-defaults  /usr/local/lib/libboost_program_options-mt.dyli
            b /usr/local/lib/libboost_system-mt.dylib /usr/local/lib/libboost_filesystem-mt.dylib
  \>> 146    Undefined symbols for architecture x86_64:
     147      "boost::system::detail::system_category_ncx()", referenced from:
     148          ___cxx_global_var_init.85 in main.cpp.o
     149          ___cxx_global_var_init.90 in main.cpp.o
     150          ___cxx_global_var_init.91 in main.cpp.o
     151          __GLOBAL__sub_I_main.cpp in main.cpp.o
     152          ___cxx_global_var_init.92 in UserDefaults.cpp.o

     ...

     156      "boost::system::detail::generic_category_ncx()", referenced from:
     157          boost::system::error_category::std_category::equivalent(int, std::__1::error_condition const&) const in main.cpp.o
     158          boost::system::error_category::std_category::equivalent(std::__1::error_code const&, int) const in main.cpp.o
     159          boost::system::error_category::std_category::equivalent(int, std::__1::error_condition const&) const in UserDefaults.cpp.o
     160          boost::system::error_category::std_category::equivalent(std::__1::error_code const&, int) const in UserDefaults.cpp.o
     161    ld: symbol(s) not found for architecture x86_64
  \>> 162    clang: error: linker command failed with exit code 1 (use -v to see invocation)
  \>> 163    make[2]: *** [dds-user-defaults/dds-user-defaults] Error 1
  \>> 164    make[1]: *** [dds-user-defaults/CMakeFiles/dds-user-defaults.dir/all] Error 2
  \>> 165    make: *** [all] Error 2
```

--- a/packages/dds/package.py
+++ b/packages/dds/package.py
@@ -56,4 +56,6 @@ class Dds(CMakePackage):
         cxxstd = self.spec.variants['cxxstd'].value
         if cxxstd != 'default':
             args.append('-DCMAKE_CXX_STANDARD=%s' % cxxstd)
+        if self.spec.satisfies('^boost@:1.69.99'):
+            args.append('-DBoost_NO_BOOST_CMAKE=ON')
         return args

--- a/packages/fairlogger/package.py
+++ b/packages/fairlogger/package.py
@@ -61,4 +61,6 @@ class Fairlogger(CMakePackage):
                        ('ON' if '+pretty' in self.spec else 'OFF'))
         if self.spec.satisfies('@1.6:'):
             args.append('-DUSE_EXTERNAL_FMT=ON')
+        if self.spec.satisfies('^boost@:1.69.99'):
+            args.append('-DBoost_NO_BOOST_CMAKE=ON')
         return args

--- a/packages/fairmq/package.py
+++ b/packages/fairmq/package.py
@@ -85,4 +85,6 @@ class Fairmq(CMakePackage):
         # NOTE Support for building the ofi transport will be added at a later
         #      point in time.
         # args.append('-DBUILD_OFI_TRANSPORT=ON')
+        if self.spec.satisfies('^boost@:1.69.99'):
+            args.append('-DBoost_NO_BOOST_CMAKE=ON')
         return args

--- a/packages/fairroot/package.py
+++ b/packages/fairroot/package.py
@@ -99,8 +99,9 @@ class Fairroot(CMakePackage):
         options.append('-DBUILD_EXAMPLES:BOOL=%s' %
                        ('ON' if '+examples' in self.spec else 'OFF'))
         options.append('-DFAIRROOT_MODULAR_BUILD=ON')
-        options.append('-DBoost_NO_SYSTEM_PATHS=TRUE')
         options.append('-DCMAKE_EXPORT_COMPILE_COMMANDS=ON')
+        if self.spec.satisfies('^boost@:1.69.99'):
+            options.append('-DBoost_NO_BOOST_CMAKE=ON')
 
         return options
 

--- a/packages/odc/package.py
+++ b/packages/odc/package.py
@@ -31,3 +31,9 @@ class Odc(CMakePackage):
     depends_on('cmake@3.12:', type='build')
     depends_on('git', type='build')
     depends_on('ninja', type='build')
+
+    def cmake_args(self):
+        args = []
+        if self.spec.satisfies('^boost@:1.69.99'):
+            args.append('-DBoost_NO_BOOST_CMAKE=ON')
+        return args

--- a/test/buildcleanup.sh
+++ b/test/buildcleanup.sh
@@ -3,7 +3,7 @@
 . test/buildsetup.sh
 
 installedMB="$(cd $HOME/install-tree && du -ms . | sed -e 's/[	 ].*$//')"
-rm -rf "$HOME/install-tree"
+# rm -rf "$HOME/install-tree"
 
 # stageKB="$(cd $HOME/stage && du -ks . | sed -e 's/[	 ].*$//')"
 (cd "$HOME" && du -sh stage)

